src/integTest/java/com/google/cloud/tools/gradle/appengine/sourcecontext/SourceContextPluginIntegrationTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2017 Google Inc. All Right Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.google.cloud.tools.gradle.appengine.sourcecontext;
+
+import com.google.common.base.Charsets;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.tools.ant.util.FileUtils;
+import org.gradle.testkit.runner.BuildResult;
+import org.gradle.testkit.runner.GradleRunner;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class SourceContextPluginIntegrationTest {
+
+  @Rule
+  public final TemporaryFolder testProjectDir = new TemporaryFolder();
+
+  public void setUpTestProject() throws IOException {
+    Path buildFile = testProjectDir.getRoot().toPath().resolve("build.gradle");
+
+    Path src = Files.createDirectory(testProjectDir.getRoot().toPath().resolve("src"));
+    InputStream buildFileContent = getClass().getClassLoader()
+        .getResourceAsStream("projects/sourcecontext-project/build.gradle");
+    Files.copy(buildFileContent, buildFile);
+
+    Path gitContext = testProjectDir.getRoot().toPath().resolve("gitContext.zip");
```

Comment:
Sweet, you got these tests to work with a zip file somehow :) I'll need to sync with you to add similar tests to the Maven plugin too.